servers/quarkus-server/src/main/resources/application.properties

```diff
@@ -85,15 +85,41 @@ nessie.version.store.type=IN_MEMORY
 #nessie.version.store.persist.cache-capacity-mb=0
 
 ## Transactional database configuration
-# Note: Nessie Quarkus Server is built with io.quarkus:quarkus-jdbc-postgresql only. If you need
-# another database, the corresponding Quarkus extensions + driver need to be included in the build
-# and the following setting updated.
+
+# Note: Nessie Quarkus Server comes with built-in support for Postgres and MariaDB, or any database
+# compatible with these. If you need another database, the corresponding Quarkus extension + driver
+# needs to be included in the build.
+# Select the datasource to use with the `nessie.version.store.persist.jdbc.datasource` property;
+# The possible built-in values are: "default" (deprecated), "postgresql" and "mariadb".
```

Comment:
Same Q as in `QuarkusJdbcConfig` ("Is it deprecated in Quarkus?").
Nessie doesn't support multiple datasources to be used in parallel/concurrently. Why not just always use the default one?

Reply:
> Why not just always use the default one?

Because `db-kind` is a build-time property. So you need to declare each datasource you intend to _maybe_ use, one by one, using different names.

Reply:
To add a bit more context about how this works:

* Quarkus can support more than one datasource, but since `db-kind` is a build-time property, all datasources and their respective extensions must be present at build time. 
* However, the `active` property is a runtime one, and can be used to selectively disable one or more datasources at runtime. This is the mechanism implemented here.
* 2 new datasources are declared with names: `postgresql` and `mariadb`. (Note that adding more datasources in the future becomes easier: "just" add the extension + one new named datasource.)
* Instead of requiring users to manually activate the datasource they want to use, I came up instead with the new option `nessie.version.store.persist.jdbc.datasource`. The`DataSourceActivator` class will lookup the datasource name and activate only that specific datasource.
* The big pain point is what to do with the "default" datasource. For now it still works as before and maps to PostgresQL, which makes upgrading easier. But my suggestion is to gradually deprecate then remove this datasource and encourage users to migrate to the new named `postgresql` datasource.
* For posterity, I also tried to implement something similar to Quarkus' `OTelFallbackConfigSourceInterceptor` (which transparently migrates legacy properties to their new names) – but datasources are really a build-time thing, and this approach did not work because the properties would be migrated at runtime – too late.

Reply:
Not great - but since that db-kind is a build-time property I don't see a better way etiher.